### PR TITLE
nrfjprog.py: Fix snr globbing to allow leading 0s

### PR DESCRIPTION
**Pull request by oyvindronningstad**
This is needed for sanitycheck hardware maps which take the serial
number directly from USB metadata.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

### CODE REVIEW

**Comment by PerMac** [comment]
Without this fix we were not able to run device-testing with hm maps on our platforms. Now the flashing works again

**Review by PerMac — APPROVED** [review]

File: scripts/west_commands/runners/nrfjprog.py [hunk 1]
```diff
@@ -83,7 +83,9 @@ def get_board_snr(self, glob):
         # multiple boards and we are connected to a terminal, in which
         # case use print() and input() to ask what the user wants.
 
-        re_glob = escape(glob).replace(r"\*", ".+")
+        # Turn glob into a regex. Also, remove leading 0s for when the snr comes
```
Comment (mbolivar-nordic) [inline scripts/west_commands/runners/nrfjprog.py:86]: Are you 100% sure this is safe? None of our boards have semantically significant leading zeros in their IDs?

Comment (oyvindronningstad) [reply scripts/west_commands/runners/nrfjprog.py]: Well, I would say 98%, but not 100% :P. Anyways, I changed it to be solved a different way now.

**Review by mbolivar-nordic — APPROVED** [review]
This doesn't break the regular workflow as far as I've tested, so no objections if it solves the sanitycheck issue.

**Pull request merged**